svelte-stuff/components/ViewStory.svelte

```diff
@@ -77,15 +77,18 @@
     {/if}
   </div>
   {#if textStory}
-    {#if !textStory.hasLiked && likeClickable && currentUserId !== textStory.creatorId}
+    {#if !textStory.hasLiked && likeClickable}
       <svg
         on:click={async () => {
           likeClickable = false;
           try {
-            await mutation(`/like-text-story/${textStory.id}`, {});
             textStory.hasLiked = true;
             textStory.numLikes++;
-          } catch {}
+            await mutation(`/like-text-story/${textStory.id}`, {});
+          } catch {
+            textStory.hasLiked = false;
+            textStory.numLikes--;
+          }
```

Comment:
I think it's not necessary to revert the ui. I would keep the mutation on top and the catch empty. Since, the mutation is really the only thing that can throw an error, it would skip 
`textStory.hasLiked = true;`
and
`textStory.numLikes++;`
anyways.
Same applies for 107-111

Reply:
The reasoning behind having hasLike and numLikes before the mutation was so that the like would show very quickly, and then let the await do its thing with the api, and if there's an error then the catch would revert those two variables back to original, while mutation.ts would send a vscode.message to the ui, but I agree is not too pretty. We just need to get rid of the ability for fast clicking all together.

So I think a better option is to start with either like/dislike enabled button, then once a user clicks a button have an async timer which disables the button until the timer is done.

Reply:
I see. I think that's causing this click issue I explained in the latest comment.
But when we add a timer, it is fixed to certain number of ms. In case the latency of this user is higher than the timer, it's over.

Reply:
I'll revert before the end of the day CST to what I had in the beginning that had the negative likes fixed for sure, we should test with realistic latency 10-20ms at most I would say, anything over that is going to be very hard not to show a delay in the UI since we're awaiting for mutation to finish.